middleware/kubernetes/kubernetes.go

```diff
@@ -384,9 +385,25 @@ func (k *Kubernetes) findPods(namespace, podname string) (pods []pod, err error)
 		return pods, nil
 	}
 
-	// TODO: implement cache verified pod responses
+	// PodModeVerified
+	podList, err := k.APIConn.podLister.List(labels.Everything())
```

Comment:
Is there a way to index this list by IP? Seems like this is going to be relatively slow for a single pod-IP lookup.